src/memo/memo.ml

```diff
@@ -147,10 +147,18 @@ let reset () =
   Caches.clear ();
   Run.restart ()
 
+module Value = struct
+  type 'a t = ('a, Exn_with_backtrace.t) Result.t
+
+  let get = function
+    | Ok a -> a
+    | Error exn -> already_reported exn
```

Comment:
"already_reported" mechanism probably doesn't really make sense anymore. We'll have to re-think that. (maybe it already didn't)

Reply:
Indeed, it appears to be implemented to support the exported `on_already_reported` which is unused, so I'll just simplify this away.